middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -1369,6 +1347,153 @@ let close_let_rec acc env ~function_declarations
+          Ident.create_local ("partial_" ^ Ident.name apply.func)
+        in
+        let closure_id =
+          Closure_id.wrap
+            (Compilation_unit.get_current_exn ())
+            (Variable.create_with_same_name_as_ident wrapper_id)
+        in
+        let params =
+          let present_args = List.length args in
+          List.mapi
+            (fun n _ ->
+              ( Ident.create_local ("param" ^ string_of_int (present_args + n)),
+                Lambda.Pgenval ))
+            missing_args
+        in
+        let return_continuation = Continuation.create ~sort:Return () in
+        let exn_continuation =
+          { apply.exn_continuation with exn_handler = Continuation.create () }
+        in
+        let all_args = args @ List.map (fun (a, _) -> IR.Var a) params in
+        let fbody acc env =
+          close_exact_or_unknown_apply acc env
+            { apply with
+              kind = Function;
+              args = all_args;
+              continuation = return_continuation;
+              exn_continuation
+            }
+            (Some approx)
+        in
+        let attr =
+          Lambda.
+            { inline = Default_inline;
+              specialise = Default_specialise;
+              local = Default_local;
+              is_a_functor = false;
+              stub = true
+            }
+        in
+        let free_idents_of_body =
+          List.fold_left
+            (fun ids -> function IR.Var id -> Ident.Set.add id ids | _ -> ids)
+            (Ident.Set.singleton apply.func)
+            all_args
+        in
+        let function_declarations =
+          [ Function_decl.create ~let_rec_ident:(Some wrapper_id) ~closure_id
+              ~kind:Lambda.Curried ~params ~return:Lambda.Pgenval
+              ~return_continuation ~exn_continuation ~body:fbody ~attr
+              ~loc:apply.loc ~free_idents_of_body ~stub:true
+              Recursive.Non_recursive ]
+        in
+        let body acc env =
+          let arg = find_simple_from_id env wrapper_id in
+          let acc, apply_cont =
+            Apply_cont_with_acc.create acc
+              ~args_approx:[Env.find_value_approximation env arg]
+              apply_continuation ~args:[arg] ~dbg:Debuginfo.none
+          in
+          Expr_with_acc.create_apply_cont acc apply_cont
+        in
+        close_let_rec acc env ~function_declarations ~body
+    in
+    match remaining_args with
+    | [] -> exact_call apply.continuation acc
+    | args ->
+      let acc, args = find_simples acc env args in
+      let wrapper_cont = Continuation.create () in
+      let _continuation = Apply.Result_continuation.Return wrapper_cont in
+      let acc, exn_continuation =
+        close_exn_continuation acc env apply.exn_continuation
+      in
+      let returned_func = Variable.create "func" in
+      let call_kind = Call_kind.indirect_function_call_unknown_arity () in
+      let inlined = LC.inlined_attribute apply.inlined in
```

Comment:
I think keeping the attribute is correct here, but it should only matter when we're not in classic mode. In classic mode the generated application will never be considered for inlining so the attribute we put here has no effect at all, but in Simplify there is a possibility that the first application is inlined, allowing its result to be resolved to a known function, which can itself be inlined. In fact `Simplify_common.split_direct_over_application` propagates the attribute too, probably for this reason.

Maybe a short comment would help ? (Oh, and to avoid confusion I would appreciate if the various definitions that are only used in the handler (from `exn_continuation` to `probe_name`) were moved into the `handler` function)